src/internal/devicestate/ledservice.go

```diff
@@ -156,3 +150,14 @@ func (l *LedService) controlLed() {
 		}
 	}
 }
+
+func (l *LedService) SetLED(ledVal int) {
```

Comment:
Don't exit from inside a method. Make a return value of type `error` and let the caller handle the error and panic if needed.

Reply:
You may be right. But this wasn't introduced by my change.

This was the old code
`		ledVal := steps[curPattern][stepIdx]
		err := l.line.SetValue(ledVal)
		if err != nil {
			log.Println(err)
			os.Exit(1)
		}`

Reply:
I know, but don't you want to leave the code better as it was before?

Reply:
Ok , fixed